ev3sources/lms2012/c_com/source/c_com.c

```diff
@@ -2252,23 +2252,20 @@ void      cComSystemCommand(RXBUF *pRxBuf, TXBUF *pTxBuf)
       pReplyListHandles->Cmd      =  LIST_OPEN_HANDLES;
       pReplyListHandles->Status   =  SUCCESS;
 
-      for (HCnt1 = 0; HCnt1 < ((MAX_FILE_HANDLES/8) + 1); HCnt1++)
+      for (NByte = 0; NByte < ((MAX_FILE_HANDLES + 7) / 8); NByte++)
       {
+        pReplyListHandles->PayLoad[NByte] = 0;
 
-        pReplyListHandles->PayLoad[HCnt1 + 2] = 0;
-
-        for(HCnt2 = 0; HCnt2 < 8; HCnt2++)
+        for (NBit = 0; NBit < 8 && (NByte * 8 + NBit) < MAX_OPEN_HANDLES; NBit++)
         {
-
-          if (0 != ComInstance.Files[HCnt2 * HCnt1].State)
+          if (0 != ComInstance.Files[NByte * 8 + NBit].State)

```

Comment:
Should use `FS_IDLE` here instead of `0`.

My personal preference would be to reverse the order as well

    if (ComInstance.Files[NByte * 8 + NBit].State != FS_IDLE)